### PR TITLE
support multiline for In tail

### LINKED ISSUE #162: in_tail multi line support
http://docs.fluentd.org/articles/in_tail

Add new UI parts for generate such config.

```
format multiline
format_firstline /^Started/
format1 /Started (?<method>[^ ]+) "(?<path>[^"]+)" for (?<host>[^ ]+) at (?<time>[^ ]+ [^ ]+ [^ ]+)\n/
format2 /Processing by (?<controller>[^\u0023]+)\u0023(?<controller_method>[^ ]+) as (?<format>[^ ]+?)\n/
format3 /(  Parameters: (?<parameters>[^ ]+)\n)?/
format4 /  Rendered (?<template>[^ ]+) within (?<layout>.+) \([\d\.]+ms\)\n/
format5 /Completed (?<code>[^ ]+) [^ ]+ in (?<runtime>[\d\.]+)ms \(Views: (?<view_runtime>[\d\.]+)ms \| ActiveRecord: (?<ar_runtime>[\d\.]+)ms\)/
```

### PR DESCRIPTION
**Pull request by yoshihara**
fixes #162 .

### CODE REVIEW

File: app/views/shared/vue/_in_tail_format.html.erb [hunk 1]
```diff
@@ -6,8 +6,21 @@
...
     <div class="form-inline form-group" v-repeat="options">
-      <label for="in_tail_setting_{{ $value }}">{{ $value }} </label>
-      <input id="in_tail_setting_{{ $value }}" type="text" name="setting[{{ $value }}]" v-model="params.setting[$value]" size="100%" class="form-control" />
+      <label for="in_tail_setting_{{ $value }}" v-if="!useTextArea">{{ $value }} </label>
+      <input id="in_tail_setting_{{ $value }}" type="{{ useTextArea ? 'hidden' : 'text' }}" name="setting[{{ $value }}]" v-model="params.setting[$value]" v-on="keyup: onKeyup" size="100%" class="form-control" />
+    </div>
+
+    <div v-if="useTextArea">
+      <div class="form-inline form-group">
+        <label for="in_tail_setting_format_firstline">format_firlstline</label>
```
Comment (uu59) [inline app/views/shared/vue/_in_tail_format.html.erb]: s/format_firlstline/format_firstline/

File: app/views/shared/vue/_in_tail_format.html.erb [hunk 2]
```diff
@@ -6,8 +6,21 @@
...
+        <input id="in_tail_setting_format_firstline" type="text" name="setting[format_firstline]" v-model="params.setting['format_firstline']" v-on="keyup: onKeyup" size="100%" class="form-control" />
+      </div>
+
+      <div class="form-group">
+        <p class="alert alert-warning"><%=t("fluentd.settings.in_tail.notice_for_multiline_limit") %></p>
```
Comment (uu59) [inline app/views/shared/vue/_in_tail_format.html.erb]: :nail_care: wanted space at `<%=t` after `=`

File: app/views/shared/vue/_in_tail_format.html.erb (hunk 1 above)
Comment (yoshihara) [inline app/views/shared/vue/_in_tail_format.html.erb]: X(

File: app/views/shared/vue/_in_tail_format.html.erb (hunk 2 above)
Comment (yoshihara) [inline app/views/shared/vue/_in_tail_format.html.erb]: oops i do that ASAP

File: lib/regexp_preview/single_line.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          }
+        },
+        matches: matches.compact,
+      }
+    end
+
+    private
+
+    def matches
+      return [] unless @regexp # such as ltsv, json, apache, etc
```
Comment (uu59) [inline lib/regexp_preview/single_line.rb]: if format is apache then `@regexp` is filled (in comment)

File: spec/lib/regexp_preview/multi_line_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+describe RegexpPreview::MultiLine do
+  describe "#matches_json" do
+    subject { RegexpPreview::MultiLine.new(File.expand_path("./spec/support/fixtures/error0.log", Rails.root), "multiline", params).matches_json }
+
+    let :params do
+      tmp = {
```
Comment (uu59) [inline spec/lib/regexp_preview/multi_line_spec.rb]: `tmp` variable name is a bit of fazzy, should be such as `request_parameters`, `req_params`, or just `params`

File: lib/regexp_preview/multi_line.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          next if lines.length < patterns.length
+          ret = detect_chunk(lines)
+          next unless ret
+          result << ret
+        end
+      end
+      result
+    end
+
+    private
```
Comment (uu59) [inline lib/regexp_preview/multi_line.rb]: `private` keyword is duplicated at L23

File: spec/lib/regexp_preview/single_line_spec.rb [hunk 6]
```diff
@@ -0,0 +1,183 @@
+require 'spec_helper'
+
+describe RegexpPreview::SingleLine do
+  describe ".initialize" do
+    subject { RegexpPreview::SingleLine.new("log_file.log", format, params) }
+
+    describe "format" do
+      let :params do
+        {
+          regexp: "(?<regexp>regexp)",
```
Comment (uu59) [inline spec/lib/regexp_preview/single_line_spec.rb]: The word `regexp` is 3 way different meaning as 1) Hash key name, 2) named capture in Regexp, and 3) will match pattern.
That is a bit of confusing, rename them possible?

File: spec/lib/regexp_preview/single_line_spec.rb [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+          time_format: "time_format",
```
Comment (uu59) [inline spec/lib/regexp_preview/single_line_spec.rb]: This line has a same issue as below :)

File: spec/lib/regexp_preview/single_line_spec.rb [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+      shared_examples "should set regexp and time_format from selected format" do
+        it do
+          expect(subject.regexp).to eq regexp
+          expect(subject.time_format).to eq time_format
+          expect(subject.params).to eq params
+        end
+      end
+
+      shared_examples "should set params only" do
+        include_examples "should set regexp and time_format from selected format" do
```
Comment (uu59) [inline spec/lib/regexp_preview/single_line_spec.rb]: :cool:

File: spec/lib/regexp_preview/multi_line_spec.rb (hunk 4 above)
Comment (yoshihara) [inline spec/lib/regexp_preview/multi_line_spec.rb]: I choose `params` thx!

File: lib/regexp_preview/multi_line.rb (hunk 5 above)
Comment (yoshihara) [inline lib/regexp_preview/multi_line.rb]: Oops!!

**Comment by uu59** [comment]
:+1:

**Pull request merged**